File: Final BMD Project/vivek/src/app/patient.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@ Injectable({
  providedIn: 'root'
})
export class PatientService {
  api = 'http://localhost:8080';

  constructor(private http : HttpClient) { }

  addPatient(patientDetails): Observable< any> {
    return this.http.post< any>(
      `${this.api}/add-patient`,
      patientDetails);
    }

    deletePatient(patient:Patient) {
      return this.http.delete< any>(`${this.api}/delete-patient/${patient.patientId}`);
      }

      getAllPatients(): Observable< any> {
        return this.http.get< any>(`${this.api}/view-allpatients`);
      }
}

File: Final BMD Project/spring boot/bookmydoctor/src/main/java/com/capgemini/bookmydoctor/service/PatientService.java
package com.capgemini.bookmydoctor.service;

import java.util.List;

import com.capgemini.bookmydoctor.dto.Patient;

public interface PatientService {
	public boolean addPatient(Patient patient);

	public Patient serarchPatient(int patientId);

	public List<Patient> getAllPatients();

	public boolean deletePatient(int patientId);
	
	public boolean modifyPatient(Patient bean);

}
